src/resmom/mom_hook_func.c

```diff
@@ -1512,8 +1514,10 @@ run_hook(hook *phook, unsigned int event_type, mom_hook_input_t *hook_input,
 				event_type == HOOK_EVENT_EXECJOB_PRETERM) {
 			int ret = 0;
 			ret = hook_env_setup(pjob, phook);
-			if ( ret != 0 )
+			if ( ret != 0 ) {
+				log_err(-1, __func__, "Unable to set the environment for the job");
```

Comment:
Use log_eventf (otherwise will log wrong error code) and log job id for which it failed